dotnet/src/Connectors/Connectors.MistralAI/Client/MistralChatChoice.cs

```diff
@@ -15,6 +17,25 @@ internal class MistralChatChoice
     [JsonPropertyName("message")]
     public MistralChatMessage? Message { get; set; }
 
+    /// <summary>
+    /// The reason the chat completion was finished.
+    /// Enum: "stop" "length" "model_length" "error" "tool_calls"
+    /// </summary>
     [JsonPropertyName("finish_reason")]
     public string? FinishReason { get; set; }
+
+    /// <summary>
+    /// Returns true if the finish reason is "tool_calls"
+    /// </summary>
+    public bool IsToolCall => this.FinishReason?.Equals("tool_calls", StringComparison.Ordinal) ?? false;
```

Comment:
If we are using this properties just within the connector those can be `internal`